### PR TITLE
Correct softmax output.

### LINKED ISSUE #343: [Bug] Multi-class probability outputs incorrect
An xgboost model with 3 classes and predict_proba=true returns incorrect output.

To fix:

The softmax output type needs to be selected in the multiclass case, like max index is selected here:
https://github.com/triton-inference-server/fil_backend/blob/956a03b9def480b7710cd2fea1f25afdf2885120/src/tl_model.h#L45

The softmax implementation also has a mistake in its indexing:
https://github.com/triton-inference-server/fil_backend/blob/956a03b9def480b7710cd2fea1f25afdf2885120/src/herring/model.hpp#L186
Should read `auto const row_begin = output + row_index * num_class;`

Also here:
https://github.com/triton-inference-server/fil_backend/blob/956a03b9def480b7710cd2fea1f25afdf2885120/src/herring/model.hpp#L199

### PR DESCRIPTION
**Pull request by RAMitchell**
Fix #343

### CODE REVIEW

**Review by wphicks — CHANGES_REQUESTED** [review]
Looks great! We do still need a non-predict_proba example in the tests, though. Could you add a new test rather than changing the old one?

**Comment by RAMitchell** [comment]
I think this is covered by xgboost_shap and lightbgm_rf, unless I've misunderstood?

**Review by wphicks — APPROVED** [review]
I guess it still is covered by xgboost_shap, but it has been removed from lightgbm_rf. Ideally we would have a non-shap example for this test, but I think it's probably not worth the extra CI time. So yes, let's get this merged! Many thanks for the fix!
